gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -846,28 +860,17 @@ public void run() {
             acquireWriteLockOnSnapshotRvv();
             try {
               if (getTxManager().getHostedTransactionsInProgress().size() == 0) {
-                if (getLoggerI18n().fineEnabled()) {
-                  getLoggerI18n().info(LocalizedStrings.DEBUG, "Clearing the Map");
-                }
                 for (Entry<String, Map<Object, BlockingQueue<RegionEntry>>> entry : oldEntryMap.entrySet()) {
-                  Map<Object, BlockingQueue<RegionEntry>> regionEntryMap = entry.getValue();
                   LocalRegion region = (LocalRegion)getRegion(entry.getKey());
-                  for (Entry<Object, BlockingQueue<RegionEntry>> oldEntry : regionEntryMap.entrySet()) {
-                    for (RegionEntry re : oldEntry.getValue()) {
-                      if (GemFireCacheImpl.hasNewOffHeap()) {
-                        // also remove reference to region buffer, if any
-                        Object value = re._getValue();
-                        if (value instanceof SerializedDiskBuffer) {
-                          ((SerializedDiskBuffer)value).release();
-                        }
-                      }
-                      // free the allocated memory
-                      if (!region.reservedTable() && region.needAccounting()) {
-                        NonLocalRegionEntry nre = (NonLocalRegionEntry)re;
-                        region.freePoolMemory(nre.getValueSize(), nre.isForDelete());
-                      }
-                    }
+                  Map<Object, BlockingQueue<RegionEntry>> regionEntryMap = entry.getValue();
+                  if (region == null) continue;
```

Comment:
SerializedDiskBuffer.release() calls have been removed? This can cause off-heap to grow too much before GC cleans it up. Typically if most data is in off-heap then heap won't be stressed and GC won't be bothered to run. So while it is not a leak, but it can cause unnecessary LMEs.

Reply:
That won't work. System.gc may not do anything. On-heap is already plagued with LME problems due to GC not bothered to run. Those issues are supposed to be gone with Off-heap but with this it will be worse than on-heap because heap won't be stressed due to off-heap lying around and GC won't run at all. Please add it back. What is really the advantage of skipping it?